skrf/network.py

```diff
@@ -958,6 +961,9 @@ def s(self, s: npy.ndarray) -> None:
         self._s = fix_param_shape(s)
         self.__generate_secondary_properties()
         self.__generate_subnetworks()
+        
+        if self.z0.ndim == 0:
+            self.z0 = self.z0
```

Comment:
this is strange. Why is it needed here?

Reply:
Consider the following example:

```python
net = rf.Network()
net.z0 = 50 # _z0 is a scalar
net.s = np.random.rand(101,2,2) # _z0 is set again from _z0
# But because _s is specifed, the _z0 shape will be broadcasted to (101,2)

```

Reply:
OK...